wnfs/src/private/directory.rs

```diff
@@ -1123,6 +1211,33 @@ impl PrivateDirectory {
         Ok(())
     }
 
+    /// Attaches a node to the specified directory without modifying the node.
+    #[allow(clippy::too_many_arguments)]
+    async fn attach_link(
+        self: &mut Rc<Self>,
+        node: PrivateNode,
+        path_segments: &[String],
+        search_latest: bool,
+        forest: &mut Rc<PrivateForest>,
+        store: &impl BlockStore,
+    ) -> Result<()> {
+        let (path, node_name) = crate::utils::split_last(path_segments)?;
+        let SearchResult::Found(dir) = self.get_leaf_dir_mut(path, search_latest, forest, store).await? else {
+            bail!(FsError::NotFound);
+        };
+
+        ensure!(
+            !dir.content.entries.contains_key(node_name),
+            FsError::FileAlreadyExists
+        );
+
+        dir.content
+            .entries
+            .insert(node_name.clone(), PrivateLink::from(node));
+
+        Ok(())
+    }
```

Comment:
Interesting use-case. So the in-memory tree structure might be good enough for fast copies but not updating the ancestry will leak secrets about unrelated directories. I'm trying to think of a way to make handle this use-case. Will need to meet with @matheus23 for this.

Reply:
Hmmm...

I want to point out that this does something to the way that permissions work in WNFS:
In WNFS, each `PrivateNode` that's stored has a bunch of `inumber`s in its accumulator (the `Namefilter` in this version).
One can think of each `inumber` as a "path segment" in a path.
You can prove (to a third party) write access to a certain `PrivateNode` by providing a signature of an `inumber`, signed by the "root owner" of a WNFS.

If you simply copy over a `PrivateRef` from one place to another, you won't be updating the whole `Namefilter` structure (this is what the `attach` logic is doing!).
Concretely this means:
If you have a file at `/Doc/Test.png` and you symlink it to `/Images/Test.png`, then someone who has write access to `/Images` *won't* be able to modify `Test.png`! (Read access will work just fine)

This breaks an invariant in WNFS: That if you have write access to a directory, you'll have write access to its contents (recursively).

I guess in theory you could fix this by including *both* `/Doc`'s and `/Images`' `inumber`s in the `Test.png` `PrivateNode`.

Can we go back to the drawing board on this? That'd probably be the "right" way of solving deduplication in rs-wnfs (at least on the file level). LMK if that makes sense to you @organizedgrime and let's brainstorm a bit about how this could look like API-wise (probably helpful if we do it live).

Reply:
> I guess in theory you could fix this by including *both* `/Doc`'s and `/Images`' `inumber`s in the `Test.png` `PrivateNode`.

I thought about this again and now I think this doesn't work (with name accumulators).